pkg/cluster/roles.go

```diff
@@ -0,0 +1,89 @@
+package cluster
+
+import (
+	"context"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateOrUpdateClusterRole creates cluster role based on define PolicyRules and optional metadata fields and updates the rules if it already exists.
+func CreateOrUpdateClusterRole(ctx context.Context, cli client.Client, name string, rules []rbacv1.PolicyRule, metaOptions ...MetaOptions) (*rbacv1.ClusterRole, error) {
+	desiredClusterRole := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Rules: rules,
+	}
+
+	if err := ApplyMetaOptions(desiredClusterRole, metaOptions...); err != nil {
+		return nil, err
+	}
+
+	foundClusterRole := &rbacv1.ClusterRole{}
+	err := cli.Get(ctx, client.ObjectKey{Name: desiredClusterRole.GetName()}, foundClusterRole)
+	if k8serr.IsNotFound(err) {
+		return desiredClusterRole, cli.Create(ctx, desiredClusterRole)
+	}
+
+	if err := ApplyMetaOptions(foundClusterRole, metaOptions...); err != nil {
+		return nil, err
+	}
+	foundClusterRole.Rules = rules
+
+	return foundClusterRole, cli.Update(ctx, foundClusterRole)
+}
+
+// DeleteClusterRole simply calls delete on a ClusterRole with the given name. Any error is returned. Check for IsNotFound.
+func DeleteClusterRole(ctx context.Context, cli client.Client, name string) error {
+	desiredClusterRole := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	return cli.Delete(ctx, desiredClusterRole)
+}
+
+// CreateOrUpdateClusterRoleBinding creates cluster role bindings based on define PolicyRules and optional metadata fields and updates the bindings if it already exists.
+func CreateOrUpdateClusterRoleBinding(ctx context.Context, cli client.Client, name string,
```

Comment:
and we still wanna keep dscinitialization.createDefaultRoleBinding() ?

Reply:
Good point, I remembered we had sth about the roles in place, but failed to find it somehow. Do you think this should be done as part of this PR?

Reply:
can leave it as a follow up.
these are new ones that are not used for now, it is fine for me to keep them as-is